### PR TITLE
Accept Store

### PR DESCRIPTION
**Pull request by wewark**
- Authorize "/store/accept/{id}": it forwards to login page if not logged in, shows the GET page iff the user is an admin, shows the forbidden page otherwise. The GET page contains the store info(only shows store name for now, no UI however) and an "accept store" button.
- Create storeRepository, storeService and storeController.
- Accept store.

Note that adding store is not implemented yet, so manually add a store to the DB for testing.

### CODE REVIEW

**Comment by SherifAbdlNaby** [comment]
Dem professional comments 💘

File: src/main/java/com/piper/valley/auth/CurrentUser.java [hunk 1]
```diff
@@ -17,7 +17,7 @@ public User getUser() {
 		return user;
 	}
 
-	public int getId() {
+	public long getId() {
```
Comment (SherifAbdlNaby) [inline src/main/java/com/piper/valley/auth/CurrentUser.java]: Why change Id type ? :?

File: src/main/java/com/piper/valley/controllers/StoreController.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+
+import java.util.Optional;
+
+@Controller
+public class StoreController {
+	/////////////////////////*  SERVICES, REPOSITORIES AND VALIDATORS SECTION  */////////////////////////////
+
```
Comment (SherifAbdlNaby) [inline src/main/java/com/piper/valley/controllers/StoreController.java]: // Move admin actions to AdminController and prefixed with /admin/{action}.
This is to collect all adminstration actions in one controller, and apply global filter over the controller itself (more secure and less scatter)

File: src/main/java/com/piper/valley/controllers/StoreController.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+			return new ModelAndView("error/404");
+		}
+
+		return new ModelAndView("store/accept", "store", store);
+	}
+
+//	@PreAuthorize("hasAuthority('ADMIN')")
+	@RequestMapping("/store/accept/{id}")
+	public ModelAndView acceptStore(@PathVariable("id") long id) {
+		storeService.acceptStore(id);
```
Comment (SherifAbdlNaby) [inline src/main/java/com/piper/valley/controllers/StoreController.java]: check store existence, if false redirect to 404 error or smth :'D

Comment (wewark) [reply src/main/java/com/piper/valley/controllers/StoreController.java]: This the POST, the store existence was checked in the GET already.

Comment (SherifAbdlNaby) [reply src/main/java/com/piper/valley/controllers/StoreController.java]: Bardo :'D, momkn y-inspect element wala 7aga, bs eshta atm it's not harmful w dah el mafrod el admin fa not critical for now.

Comment (wewark) [reply src/main/java/com/piper/valley/controllers/StoreController.java]: El mfrod el admin byfham 😂

**Pull request merged**